librz/bin/format/le/le.c

```diff
@@ -1,10 +1,40 @@
 // SPDX-FileCopyrightText: 2019 GustavoLCR <gugulcr@gmail.com>
+// SPDX-FileCopyrightText: 2023 svr <svr.work@protonmail.com>
 // SPDX-License-Identifier: LGPL-3.0-only
 
 #include "le.h"
 #include <rz_bin.h>
+#include <rz_types.h>
+#include <sdbht.h>
 
-const char *le_get_module_type(rz_bin_le_obj_t *bin) {
+// Sources:
+//
+// [1] IBM OS/2 16/32-BIT OBJECT MODULE FORMAT (OMF) AND LINEAR EXECUTABLE MODULE FORMAT (LX) rev10:
+//     http://www.edm2.com/index.php/IBM_OS/2_16/32-bit_Object_Module_Format_%28OMF%29_and_Linear_eXecutable_Module_Format_%28LX%29
+//
+// [2] lxLite LX executable packer:
+//     https://github.com/bitwiseworks/lxlite/blob/master/src/os2exe.pas
+//
+// [3] DOS/32 Advanced DOS Extender unbind utility:
+//     https://github.com/abbec/dos32a/blob/master/src/sb/sbind.asm
+//
+
+#define CHECK(expr) \
+	if (!(expr)) { \
+		goto fail_cleanup; \
+	}
+
+#define CHECK_READ(X, tmp, out) \
+	CHECK(rz_buf_read##X##_offset(buf, offset, &tmp) && *offset <= offset_end) \
+	out = tmp;
+
+#define CHECK_READ8(out)  CHECK_READ(8, tmp8, out)
+#define CHECK_READ16(out) CHECK_READ(_le16, tmp16, out)
+#define CHECK_READ32(out) CHECK_READ(_le32, tmp32, out)
+
+// --- Auxilliary functions -----------------------------------------------------------------------
```

Comment:
Same, use Doxygen to split categories

Reply:
Do you know the syntax by chance?

Reply:
See https://www.doxygen.nl/manual/commands.html